api/lambdas/sparql.ts

```diff
@@ -0,0 +1,99 @@
+import got from 'got';
+import type {
+  APIGatewayProxyEventV2,
+  APIGatewayProxyResult,
+} from 'aws-lambda';
+import { URL } from 'url';
+import { pick } from 'lodash';
+
+export const getStringFromEnv = (key: string): string | Error =>
+  process.env[key] || new Error(`${key} not set`);
+
+export const okResponse = (
+  contentType: string,
+  body: string
+): APIGatewayProxyResult => ({
+  statusCode: 200,
+  headers: { 'Content-Type': contentType },
+  body,
+});
+
+export const badRequestResponse = (
+  body: string,
+  contentType?: string
+): APIGatewayProxyResult => ({
+  statusCode: 400,
+  headers: { 'Content-Type': contentType ?? 'text/plain' },
+  body,
+});
+
+export const internalServerErrorResponse: APIGatewayProxyResult = {
+  statusCode: 500,
+  headers: { 'Content-Type': 'text/plain' },
+  body: 'Internal Server Error',
+};
+
+export const badGatewayResponse: APIGatewayProxyResult = {
+  statusCode: 502,
+  headers: { 'Content-Type': 'text/plain' },
+  body: 'Bad Gateway',
+};
+
+export type PostSparqlEvent = Pick<APIGatewayProxyEventV2, 'body'>;
+
+export const handler = async (
+  event: PostSparqlEvent
+): Promise<APIGatewayProxyResult> => {
+  const { body: query } = event;
+
+  const sparqlUrlString = getStringFromEnv('SPARQL_URL');
+
+  if (sparqlUrlString instanceof Error) {
+    console.log('SPARQL_URL not set');
+    return internalServerErrorResponse;
+  }
+
+  let sparqlUrl: URL;
+  try {
+    sparqlUrl = new URL(sparqlUrlString);
+  } catch {
+    console.log(`Invalid SPARQL_URL: ${sparqlUrlString}`);
+    return internalServerErrorResponse;
+  }
+
+  if (!query) {
```

Comment:
Not a big deal but is it worth doing this check on line 48 - immediately after you deconstruct the event?

Reply:
Changed in 0647279